### ISSUE TITLE
GUI always can not start

### ISSUE THREAD

**Issue by TwelveNight** [0bCdian/Waypaper-Engine#4]
Can not start the GUI from the tray,When I config the app start from the tray,I can never start the GUI

**Comment by 0bCdian** [comment]
What window manager / bar are you using, I tested this on sway and hyprland, using waybar, eww and sway's default bar and didn't have any more issues, try running from the terminal to see what error is popping up so I can dig into it.

**Comment by TwelveNight** [comment]
I use hyprland and ags which something like eww,inspired by you，I test is at waybar, it can work correctly,may be the ags problem?Should I make a issuse at  [ags](https://github.com/Aylur/ags)

**Comment by TwelveNight** [comment]
I 

> I use(使用) hyprland ,and for bar I use(使用) ags that something like eww,inspired(激发) by you，I test is at waybar, it can work(工作) correctly(正确),may(可能) be the ags problem?Should I make a issuse at [ags](https://github.com/Aylur/ags)?我用hyprland，而bar我用ags，类似eww，受你的启发，我在waybar测试，它可以正常工作，可能是ags问题？我应该在ags上提出问题吗？

I find in ags that left-click and right-click behave the same
![image](https://github.com/0bCdian/Waypaper-Engine/assets/113519679/80c9045a-1b50-4ac6-a542-2147d8f5af62)

**Comment by TwelveNight** [comment]
> I 我
> 
> > I use(使用)(使用) hyprland ,and for bar I use(使用)(使用) ags that something like eww,inspired(激发)(激发) by you，I test is at waybar, it can work(工作)(工作) correctly(正确)(正确),may(可能)(可能) be the ags problem?Should I make a issuse at [ags](https://github.com/Aylur/ags)?我用hyprland，而bar我用ags，类似eww，受你的启发，我在waybar测试，它可以正常工作，可能是ags问题？我应该在ags上提出问题吗？
> 
> I find(找到) in ags that left-click and right-click behave(行为) the same我发现 ags 中左键单击和右键单击的行为相同 ![image](https://private-user-images.githubusercontent.com/113519679/290965019-80c9045a-1b50-4ac6-a542-2147d8f5af62.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI2OTI2NjIsIm5iZiI6MTcwMjY5MjM2MiwicGF0aCI6Ii8xMTM1MTk2NzkvMjkwOTY1MDE5LTgwYzkwNDVhLTFiNTAtNGFjNi1hNTQyLTIxNDdkOGY1YWY2Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIxNlQwMjA2MDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZGM2OGI3ZmIwMzlkZjM2MWVhYzZhZGRiNDdmYTE2ZDgxNjBmMTNhYjZjOWRhOWE5Y2M2NmNhYWE4ODE3NzcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.N0Roj22prZPDrElhJpJoLTtxr-EsX3BVmmRrF3lFGy0)

Ok,I belive is this problem,the other app the same behave

**Comment by 0bCdian** [comment]
I haven't tested ags really, but the same behavior is present in eww (right and left click doing the same thing), I submitted an issue to eww but haven't gotten a response yet, and tested this with other apps that use the tray and all behave the same way, so I suspect this is more of an issue of eww/ags with their tray implementation than anything else. I retested on hyprland and sway with waybar and didn't have problems starting from the tray. I will test as soon as I can with ags to see if there are any crashes.
